homeassistant/components/nextcloud/config_flow.py

```diff
@@ -0,0 +1,69 @@
+"""Config flow to configure the Nextcloud integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from nextcloudmonitor import NextcloudMonitor, NextcloudMonitorError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from . import DOMAIN
+
+DATA_SCHEMA_USER = vol.Schema(
+    {
+        vol.Required(CONF_URL): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class NextcloudConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a Nextcloud config flow."""
+
+    VERSION = 1
+
+    async def async_step_import(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by configuration file."""
+        if user_input is not None:
+            self._async_abort_entries_match({CONF_URL: user_input.get(CONF_URL)})
+        return await self.async_step_user(user_input)
```

Comment:
If the connection fails during import, this would leave an incomplete flow.
I'm not sure if this is intended - maybe it should bail out instead?